scripts/pi-hole/js/queries.js

```diff
@@ -496,28 +432,108 @@ $(function () {
-  input.setAttribute("autocorrect", "off");
-  input.setAttribute("autocapitalize", "off");
-  input.setAttribute("spellcheck", false);
-
-  var chkboxData = localStorage.getItem("query_log_filter_chkbox");
-  if (chkboxData !== null) {
-    // Restore checkbox state
-    $("#autofilter").prop("checked", chkboxData === "true");
+  if (input !== null) {
+    input.setAttribute("autocomplete", "off");
+    input.setAttribute("autocorrect", "off");
+    input.setAttribute("autocapitalize", "off");
+    input.setAttribute("spellcheck", false);
+  }
+});
+
+function addColumnFilter(event, colID, filterstring) {
+  // Don't do anything when NOT explicitly requesting multi-selection functions
+  if (!event.ctrlKey && !event.metaKey && !event.shiftKey) {
+    return;
+  }
+
+  if (event.shiftKey) {
+    filterstring = "";
+  }
+
+  tableFilters[colID] = filterstring;
+
+  applyColumnFiltering();
+}
+
+function resetColumnsFilters() {
+  tableFilters.forEach(function (value, index) {
+    tableFilters[index] = "";
+  });
+
+  // Clear filter reset button
+  applyColumnFiltering();
+}
+
+function applyColumnFiltering() {
+  var showReset = false;
+  tableFilters.forEach(function (value, index) {
+    // Prepare regex filter string
+    var regex = "";
+
+    // Split filter string if we received a combined ID#Name column
+    var valArr = value.split("#");
+    if (valArr.length > 0) {
+      value = valArr[0];
+    }
+
+    if (value.length > 0) {
+      // Exact matching
+      regex = "^" + value + "$";
+
+      // Add background color
+      tableApi.$("td:eq(" + index + ")").addClass("highlight");
+
+      // Remember to show reset button
+      showReset = true;
+    } else {
+      // Clear background color
+      tableApi.$("td:eq(" + index + ")").removeClass("highlight");
+    }
+
+    // Apply filtering on this column (regex may be empty -> no filtering)
+    tableApi.column(index).search(regex, true, true);
+  });
+
+  if (showReset) {
+    showResetButton();
   } else {
-    // Initialize checkbox
-    $("#autofilter").prop("checked", true);
-    localStorage.setItem("query_log_filter_chkbox", true);
+    hideResetButton();
   }
 
-  $("#autofilter").click(function () {
-    localStorage.setItem("query_log_filter_chkbox", $("#autofilter").prop("checked"));
+  // Trigger table update
+  tableApi.draw();
+}
+
+function showResetButton() {
+  var button = $("#resetButton");
+  var text = "";
+  tableFilters.forEach(function (value, index) {
+    // Split filter string if we received a combined ID#Name column
+    var valArr = value.split("#");
+    if (valArr.length > 1) {
+      value = valArr[1];
+    }
+
+    // Create or add to button text
+    if (value.length > 0 && text.length === 0) {
+      text = "Clear filtering on " + colTypes[index] + ' "' + value + '"';
+    } else if (value.length > 0) {
+      text += " and " + colTypes[index] + ' "' + value + '"';
+    }
   });
-});
+
+  button.text(text);
```

Comment:
Did you check if this looks good with long content? Maybe we shouldn't change the button's text, and do it with a span above it or something?

Reply:
https://discourse.pi-hole.net/t/explicit-search-in-pi-hole-admin-console/28893/57?u=yubiuser

Reply:
I think probably a "Clear current filter" would suffice. That is what it is doing, after all.